Workflow/Sources/ApplyContext.swift

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/// Internal utility protocol so that `WorkflowTesting` can provide an alternate implementation
+protocol ApplyContextType<WorkflowType> {
```

Comment:
i'm no longer certain this abstraction is needed. i mostly copied it from the analogous thing done for RenderTester, but unlike in that case i don't think the testing infra relies on 'real' internals to apply actions.